### PR TITLE
Refactor plugin to work when multiple TS projects are loaded

### PR DESCRIPTION
**Pull request by alangpierce**
This is pretty much a rewrite of the plugin to better handle situations where a
single editor is working with files from multiple TS projects. Previously, the
init and create functions were being called once per project, so when loading
the second project we'd register a new IPC listener and clobber the old one, so
any file requests received from the wrong project would get a "file not found"
error and crash the TSStatusClient.

After this rewrite, we bookkeep the open projects and register a single IPC
listener that tries each project and combines the results. I also refactored it
into two classes and smaller functions and changed the logging system to just
append to a hard-coded file.

This plugin is still non-ideal for multi-project use cases because if you happen
to have the projects open in different editors, they'll run in different TS
server processes and the second listener will clobber the first. There doesn't
seem to be an obvious way to configure node-ipc to have multiple servers and get
a response from each of them, but I'm sure there's a way to make it work.

Test Plan:
Enable logging and confirm
Tweak the code so that the "not found in any project" message always appears and
make sure it that it looks reasonable.
Try using `bin/dev check` in the aurelia project with PyCharm and with VSCode
and ensure that both now work.

### CODE REVIEW

**Comment by alangpierce** [comment]
@cbuchan @damondoucet this is our public open source TS server plugin that powers `bin/dev check`, would be good to get you two up to speed on it so I'm putting you both as reviewers (but feel free to delegate/reshuffle). It was crashing when doing Cypress work because browser-tests is its own TS project, so this should fix that issue.

**Comment by damondoucet** [comment]
oooh interesting! Adding @westrik on this.

@cbuchan I think at least two people on our team should have context on this, and I probably shouldn't be one of them given I'm usually too busy to assist. Feel free to move from yourself to someone else!

**Comment by damondoucet** [comment]
womp apparently I can't change the reviewers? @alangpierce can you replace me with @westrik for now?

**Comment by alangpierce** [comment]
@damondoucet sorry, the github permissions need updating, it just has me as admin and everyone else as read. I'll make you and @cbuchan admins as well and set everyone else to maintain. I'll also set you and Collin as maintainers on the npm package.

**Comment by alangpierce** [comment]
@damondoucet done, can you try now?

**Comment by damondoucet** [comment]
Worked, thanks!

File: tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "downlevelIteration": true,
     "esModuleInterop": true,
     "importHelpers": true,
-    "lib": ["es2017", "DOM"],
+    "lib": ["es2020"],
```
Comment (westrik) [inline tsconfig.json:7]: OOC why was this change necessary? (Or is this just an unrelated improvement?)

**Review by westrik — APPROVED** [review]
Looks good to me — thanks for making this change! It will make working on multi-project changes a lot easier.

To test this, I first updated aurelia's `package.json` to use `https://github.com/benchling/ts-expose-status-plugin.git#08cb4ce` instead of `^1.1.0` (and ran `yarn` + `cd node_modules/ts-expose-status-plugin; yarn build`). I then edited files from multiple TypeScript projects in PyCharm and committed those changes. Finally I ran `bin/dev check`, which completed with no errors.

File: tsconfig.json (hunk 1 above)
Comment (alangpierce) [reply tsconfig.json:7]: It was needed for `flatMap`.

**Comment by alangpierce** [comment]
@westrik 

> To test this, I first updated aurelia's package.json to use https://github.com/benchling/ts-expose-status-plugin.git#08cb4ce instead of ^1.1.0 (and ran yarn + cd node_modules/ts-expose-status-plugin; yarn build). I then edited files from multiple TypeScript projects in PyCharm and committed those changes. Finally I ran bin/dev check, which completed with no errors.

Thanks for getting that figured out! An easier way that I've been doing is to clone the project, then `yarn link` from the `ts-expose-status-plugin` directory, then `yarn link ts-expose-status-plugin` from the aurelia repo, which sets up a symlink from node_modules to the separate checkout. I think that's the most common way to test unpublished changes in the node world. You still need to `yarn build` from `ts-expose-status-plugin` and restart the editor TS service to pick up any changes.

**Review by cbuchan — APPROVED** [review]
Thank you for working on this! 🙏
